File: ios/DemoExpoModule.swift
import ExpoModulesCore

public class DemoExpoModule: Module {
  public func definition() -> ModuleDefinition {
    Name("DemoExpo")

    Class(DemoExpoModuleSharedObject.self) {
      Constructor { () -> DemoExpoModuleSharedObject in
        return DemoExpoModuleSharedObject()
      }

      Property("count") { (ref: DemoExpoModuleSharedObject) -> Int in
        return ref.count
      }
      .set { (ref: DemoExpoModuleSharedObject, count: Int) in
        ref.count = count
      }
    }
  }
}

File: android/src/main/java/expo/modules/demoexpo/DemoExpoModule.kt
package expo.modules.demoexpo

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class DemoExpoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("DemoExpo")

    Class(DemoExpoModuleSharedObject::class) {
      Constructor {
        val instance = DemoExpoModuleSharedObject(appContext)
        return@Constructor instance
      }

      Property("count")
        .get { ref: DemoExpoModuleSharedObject ->
          ref.count
        }
        .set { ref: DemoExpoModuleSharedObject, count: Int ->
          ref.count = count
        }
    }
  }
}
